solarforecastarbiter/reference_forecasts/persistence.py

```diff
@@ -362,3 +365,244 @@ def _check_intervals_times(interval_label, data_start, data_end,
+        interval_label is *ending* or *instant*.
+    interval_length : pd.Timedelta
+        Forecast interval length
+    interval_label : str
+        instant, beginning, or ending
+    load_data : function
+        A function that loads the observation data. Must have the
+        signature load_data(observation, data_start, data_end) and
+        properly account for observation interval label.
+    axis : {'x', 'y'}
+        The axis on whicht he constant values of the CDF is specified. The axis
+        can be either *x* (constant variable values) or *y* (constant
+        percentiles).
+    constant_values : array_like
+        The variable values or percentiles.
+
+    Returns
+    -------
+    forecasts : list of pd.Series
+        The persistence forecasts, returned in the same order as
+        *constant_values*. If *axis* is *x*, the forecast values are
+        percentiles (e.g. 25%). If instead *axis* is *y*, the forecasts values
+        have the same units as the observation data (e.g. MW).
+
+    Raises
+    ------
+    ValueError
+        If there is insufficient data for matching by time of day or the
+        **axis** parameter is invalid.
+
+    Notes
+    -----
+    Assumes that there is at least 20 days of *observation* data available
+    based on [1]_, [2]_, [3]_.
+
+    References
+    ----------
+    .. [1] Allessandrini et al. (2015) "An analog ensemble for short-term
+       probabilistic solar power forecast", Appl. Energy 157, pp. 95-110.
+       doi: 10.1016/j.apenergy.2015.08.011
+    .. [2] Yang (2019) "A universal benchmarking method for probabilistic
+       solar irradiance forecasting", Solar Energy 184, pp. 410-416.
+       doi: 10.1016/j.solener.2019.04.018
+    .. [3] Doubleday, Van Scyoc Herndandez and Hodge (2020) "Benchmark
+       probabilistic solar forecasts: characteristics and recommendations",
+       Solar Energy 206, pp. 52-67. doi: 10.1016/j.solener.2020.05.051
+
+    See also
+    --------
+    :py:func:`solarforecastarbiter.reference_forecasts.persistence.persistence_probabilistic`
+
+    """
+    # ensure that we're using times rounded to multiple of interval_length
+    _check_intervals_times(observation.interval_label, data_start, data_end,
+                           forecast_start, forecast_end,
+                           observation.interval_length)
+
+    closed = datamodel.CLOSED_MAPPING[interval_label]
+    fx_index = pd.date_range(start=forecast_start, end=forecast_end,
+                             freq=interval_length, closed=closed)
+
+    # observation data resampled to match forecast sampling
+    obs = load_data(observation, data_start, data_end)
+    obs = obs.resample(interval_length, closed=closed).mean()
+
+    # time of day: minutes past midnight (e.g. 0=12:00am, 75=1:15am)
+    obs_timeofday = (obs.index.hour * 60 + obs.index.minute).astype(int)
+    fx_timeofday = (fx_index.hour * 60 + fx_index.minute).astype(int)
```

Comment:
need to be careful here about timezone. If somehow forecast_start/end have a different timezone from obs, matching by time of day might use the wrong data. 

```suggestion
    if fx_index.tzinfo is None:
        fx_index = fx_index.tz_localize(obs.index.tzinfo)
    else:
        fx_index = fx_index.tz_convert(obs.index.tzinfo)
    fx_timeofday = (fx_index.hour * 60 + fx_index.minute).astype(int)
```

Reply:
Revised as suggested.

Reply:
I'll add a test for code coverage

Reply:
I've added a test for this.

Reply:
@alorenzo175 @wholmgren Actually, do we actually want the forecasts returned with the same timezone as the observations? Does that match the logic/expectations in the other reference forecasts? I don't see any `tz_convert` operations in `persistence.py`.

Reply:
I don't think it matters much, but yes, to be consistent with ``persistence.py`` maybe convert the observations to the forecast index timezone if not none, otherwise use the observation timezone

Reply:
Sounds good. I've removed the `tz_convert` and just kept the `tz_localize` for if the fx is missing tz info.

Reply:
I think you still need a `tz_convert` for obs. For example, if the say `forecast_start` is specified as a PST timestamp, but the observations are returned as UTC. Then the `fx_index` will be PST but `obs.index` will be UTC and the `obs_timeofday` and `fx_timeofday` will be shifted. I think something like this will take care of it
```python
if obs.index.tzinfo is not None:
   obs = obs.tz_convert(fx_index.tzinfo)
else:
   obs = obs.tz_localize(fx_index.tzinfo)
```
which needs to go after fx_index is localized but before the obs_timeofday. The other persistence functions don't have a `tz_convert` because they aren't comparing time of day, although they might have issues if something is unlocalized.

Reply:
You're correct. (I just wrote a comment saying something similar before I saw your reply.)

Reply:
With e4dd65f, the `persistence_probabilistic_timeofday()` should now properly handle all combinations of obs and fx timezones.